packages/core/data-transfer/src/strapi/providers/utils.ts

```diff
@@ -221,6 +226,32 @@ export const connectToWebsocket = (address: Address, options?: Options): Promise
       );
     });
 
+    server.on('message', (raw: RawData) => {
+      const response: Server.Message = JSON.parse(raw.toString());
+      if (response.diagnostic) {
+        if (response.diagnostic.kind === 'info') {
+          diagnostics?.report({
+            details: response.diagnostic.details,
+            kind: response.diagnostic.kind,
+          });
+        }
+
+        if (response.diagnostic.kind === 'warning') {
+          diagnostics?.report({
+            details: response.diagnostic.details,
+            kind: response.diagnostic.kind,
+          });
+        }
+
+        if (response.diagnostic.kind === 'error') {
+          diagnostics?.report({
+            details: response.diagnostic.details,
+            kind: response.diagnostic.kind,
+          });
```

Comment:
Since we're doing the same thing for info, warning and error, couldn't we factorize this a bit ?